packages/schematics/angular/utility/workspace_spec.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { EmptyTree, SchematicContext } from '@angular-devkit/schematics';
+import { getWorkspace as readWorkspace, updateWorkspace, writeWorkspace } from './workspace';
+
+const TEST_WORKSPACE_CONTENT = JSON.stringify({
+  version: 1,
+  projects: {
+    'test': {},
+  },
+});
+
+describe('readWorkspace', () => {
+  it('reads a workspace using the default path value', async () => {
+    const tree = new EmptyTree();
+    tree.create('/angular.json', TEST_WORKSPACE_CONTENT);
+
+    const workspace = await readWorkspace(tree);
+    expect(workspace.projects.has('test')).toBeTrue();
+  });
+
+  it('reads a workspace when specifying a directory path', async () => {
+    const tree = new EmptyTree();
+    tree.create('/xyz/angular.json', TEST_WORKSPACE_CONTENT);
+
+    const workspace = await readWorkspace(tree, '/xyz/');
+    expect(workspace.projects.has('test')).toBeTrue();
+  });
+
+  it('reads a workspace when specifying a file path', async () => {
+    const tree = new EmptyTree();
+    tree.create('/angular.json', TEST_WORKSPACE_CONTENT);
+
+    const workspace = await readWorkspace(tree, 'angular.json');
```

Comment:
Nit: Should use a different file path here as `readWorkspace()` could ignore this field altogether and still pass the test because it happens to match the default.